Sprint-1/fix/median.js

```diff
@@ -3,9 +3,26 @@
 // If you're in the Sprint-1 directory, you can run `npm test -- fix` to run the tests in the fix directory
 
 function calculateMedian(list) {
-  const middleIndex = Math.floor(list.length / 2);
-  const median = list.splice(middleIndex, 1)[0];
-  return median;
+
+  // 〰️ Create a copy of the input list to avoid modifying the original array
+  const copiedList = [...list];
+
```

Comment:
The arrays specified in `median.test.js` happened to be sorted.

Normally, before we can easily find the median value in an array, we need to sort the array first.
Can you try inserting a statement to sort the numbers in `copiedList` first?

Reply:
You’re absolutely right. I missed that. Finding the median assumes the array is sorted, and the test cases in median.test.js were already sorted by coincidence. To handle cases where the input array isn't sorted, I’ve added a statement to sort copiedList before calculating the median.

Thank you for bringing this up! It was an important detail to address, and this change makes the function more reliable in all scenarios.